src/Framework/Framework/Binding/ExtractGenericArgumentDataContextChangeAttribute.cs

```diff
@@ -24,7 +25,7 @@ public ExtractGenericArgumentDataContextChangeAttribute(Type genericType, int ty
             throw new ArgumentOutOfRangeException(nameof(typeArgumentIndex), $"The {nameof(typeArgumentIndex)} is not a valid index of a type argument!");
         }
 
-        GenericType = genericType;
+        GenericType = new ResolvedTypeDescriptor(genericType);
         TypeArgumentIndex = typeArgumentIndex;
         Order = order;
     }
```

Comment:
I think you'll need add `ExtractGenericArgumentDataContextChangeAttribute(ITypeDescriptor genericType, int typeArgumentIndex, int order = 0)` overload